jina/executors/indexers/vector.py

```diff
@@ -154,18 +157,27 @@ def raw_ndarray(self) -> Optional['np.ndarray']:
             return self._load_gzip(self.index_abspath)
         elif self.size is not None and os.stat(self.index_abspath).st_size:
             self.logger.success(f'memmap is enabled for {self.index_abspath}')
-            return np.memmap(self.index_abspath, dtype=self.dtype, mode='r', shape=(self.size, self.num_dim))
+            # `==` is required. `is False` does not work in np
+            deleted_keys = len(self.existing_keys[self.existing_keys == False]) # noqa
+            return np.memmap(self.index_abspath, dtype=self.dtype, mode='r',
+                             shape=(self.size + deleted_keys, self.num_dim))
 
     def query_by_id(self, ids: Union[List[int], 'np.ndarray'], *args, **kwargs) -> 'np.ndarray':
         int_ids = [self.ext2int_id[j] for j in ids]
-        return self.raw_ndarray[int_ids]
+        # TODO discussion on what we want to return on key that has been deleted
```

Comment:
It should have the same behavior, as if we would query for a key that is not present.

Reply:
Atm that is not handled. On master it simply raises

```
        int_ids = [self.ext2int_id[j] for j in ids]
```

Is that what we want?

Reply:
OK, to be fair, I don't see the use-case of that function so I cannot give you a good answer. But it seems like we should raise an `AttributeError` until the use-case is understood. Please check, if the exception is catched in the calling function and if you could find a pattern, that fits. Thanks!

Reply:
Shouldn't it be `KeyError`? I guess it is similar in logic to how in the `PbIndexer` we handle missing keys on update/delete https://github.com/jina-ai/jina/blob/4ff025588fcfdb46a41bb011f260b1a53217c3c5/jina/executors/indexers/keyvalue.py#L82

Reply:
Yes you are absolutely right. :)